File: libs/ontology-kernel/src/models/function_authoring.rs
use serde::{Deserialize, Serialize};

use super::function_package::FunctionCapabilities;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FunctionAuthoringTemplate {
    pub id: String,
    pub runtime: String,
    pub display_name: String,
    pub description: String,
    pub entrypoint: String,
    pub starter_source: String,
    pub default_capabilities: FunctionCapabilities,
    pub recommended_use_cases: Vec<String>,
    pub cli_scaffold_template: Option<String>,
    pub sdk_packages: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FunctionSdkPackageReference {
    pub language: String,
    pub path: String,
    pub package_name: String,
    pub generated_by: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FunctionAuthoringSurfaceResponse {
    pub templates: Vec<FunctionAuthoringTemplate>,
    pub sdk_packages: Vec<FunctionSdkPackageReference>,
    pub cli_commands: Vec<String>,
}

File: libs/ontology-kernel/models/function_authoring.go
package models

// FunctionAuthoringTemplate mirrors `struct FunctionAuthoringTemplate`
// in `libs/ontology-kernel/src/models/function_authoring.rs`.
type FunctionAuthoringTemplate struct {
	ID                   string               `json:"id"`
	Runtime              string               `json:"runtime"`
	DisplayName          string               `json:"display_name"`
	Description          string               `json:"description"`
	Entrypoint           string               `json:"entrypoint"`
	StarterSource        string               `json:"starter_source"`
	DefaultCapabilities  FunctionCapabilities `json:"default_capabilities"`
	RecommendedUseCases  []string             `json:"recommended_use_cases"`
	CLIScaffoldTemplate  *string              `json:"cli_scaffold_template"`
	SDKPackages          []string             `json:"sdk_packages"`
}

// FunctionSDKPackageReference mirrors `struct FunctionSdkPackageReference`.
type FunctionSDKPackageReference struct {
	Language    string `json:"language"`
	Path        string `json:"path"`
	PackageName string `json:"package_name"`
	GeneratedBy string `json:"generated_by"`
}

// FunctionAuthoringSurfaceResponse mirrors
// `struct FunctionAuthoringSurfaceResponse`.
type FunctionAuthoringSurfaceResponse struct {
	Templates   []FunctionAuthoringTemplate   `json:"templates"`
	SDKPackages []FunctionSDKPackageReference `json:"sdk_packages"`
	CLICommands []string                      `json:"cli_commands"`
}
